analytics_data_api/v0/views/learners.py

```diff
@@ -352,6 +355,22 @@ def get_queryset(self):
         return queryset
 
 
+class EnterpriseLearnerEngagementView(generics.ListAPIView):
+    """
+    Return engagement data for enterprise learners.
+    """
+    serializer_class = EnterpriseLearnerEngagementSerializer
+    pagination_class = EdxPaginationSerializer
+
+    def get_queryset(self):
+        enterprise_users = EnterpriseUser.objects.filter(
+            enterprise_id=self.kwargs.get('enterprise_customer')
+        ).values_list(
+            'user_username', flat=True
+        )
+        return ModuleEngagement.objects.filter(username__in=enterprise_users)
```

Comment:
is there any way we can avoid this `__in` operation? [might be costly](https://dba.stackexchange.com/questions/207255/performance-mysql-id-vs-id-in-multiple-values)